azurerm/internal/services/policy/resource_arm_policy_remediation.go

```diff
@@ -0,0 +1,248 @@
+	}
+	if resp.ID == nil {
+		return fmt.Errorf("cannot read Policy Remediation %q (Scope %q) ID", name, scope.ScopeId())
+	}
+	d.SetId(*resp.ID)
+
+	return resourceArmPolicyRemediationRead(d, meta)
+}
+
+func resourceArmPolicyRemediationRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Policy.RemediationsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.PolicyRemediationID(d.Id())
+	if err != nil {
+		return fmt.Errorf("unable to read Policy Remediation: %+v", err)
+	}
+
+	resp, err := RemediationGetAtScope(ctx, client, id.Name, id.PolicyScopeId)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Policy Remediation %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("unable to read Policy Remediation %q (Scope %q): %+v", id.Name, id.ScopeId(), err)
+	}
+
+	d.Set("name", id.Name)
+	d.Set("scope", id.ScopeId())
+
+	if props := resp.RemediationProperties; props != nil {
+		if err := d.Set("location_filters", flattenArmRemediationLocationFilters(props.Filters)); err != nil {
+			return fmt.Errorf("unable to set `location_filters`: %+v", err)
+		}
+
+		d.Set("policy_assignment_id", props.PolicyAssignmentID)
+		d.Set("policy_definition_reference_id", props.PolicyDefinitionReferenceID)
+	}
+
+	return nil
+}
+
+func resourceArmPolicyRemediationDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Policy.RemediationsClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.PolicyRemediationID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	switch scope := id.PolicyScopeId.(type) {
+	case parse.ScopeAtSubscription:
+		_, err = client.DeleteAtSubscription(ctx, scope.SubscriptionId, id.Name)
+	case parse.ScopeAtResourceGroup:
+		_, err = client.DeleteAtResourceGroup(ctx, scope.SubscriptionId, scope.ResourceGroup, id.Name)
+	case parse.ScopeAtResource:
+		_, err = client.DeleteAtResource(ctx, scope.ScopeId(), id.Name)
+	case parse.ScopeAtManagementGroup:
+		_, err = client.DeleteAtManagementGroup(ctx, scope.ManagementGroupId, id.Name)
+	default:
+		return fmt.Errorf("unable to delete Policy Remediation %q: invalid scope type", id.Name)
+	}
+	if err != nil {
+		return fmt.Errorf("unable to delete Policy Remediation %q (Scope %q): %+v", id.Name, id.ScopeId(), err)
+	}
+
+	return nil
+}
+
+func expandArmRemediationLocationFilters(input []interface{}) *policyinsights.RemediationFilters {
+	if len(input) == 0 {
+		return nil
+	}
+
+	result := policyinsights.RemediationFilters{
+		Locations: utils.ExpandStringSlice(input),
+	}
+
+	return &result
+}
+
+func flattenArmRemediationLocationFilters(input *policyinsights.RemediationFilters) []interface{} {
+	if input == nil {
+		return make([]interface{}, 0)
+	}
+
+	return utils.FlattenStringSlice(input.Locations)
+}
```

Comment:
i think for both these functions we could just inline the expand/flatten slice functions?

Reply:
Emm, in my memory, the inline call was the first version of this, and then I changed them into this... Let me check if there is problems when changing these back

Reply:
Done